contracts/tips/src/msg.rs

```diff
@@ -0,0 +1,282 @@
+use crate::error::ContractError;
+use crate::state::StateServiceFee;
+use cosmwasm_std::{Addr, Coin, Uint128, Uint64};
+use schemars::JsonSchema;
+use serde::{Deserialize, Serialize};
+
+/// Fees required to execute [`ExecuteMsg::SendTip`].
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema)]
+#[serde(rename_all = "snake_case")]
+pub enum ServiceFee {
+    /// Represents a fixed fee that is deducted from the tip.
+    Fixed {
+        /// Coins that are deducted from the tip.
+        amount: Vec<Coin>,
+    },
+    /// Represents a percentage that is deducted from the tip.
+    Percentage {
+        /// Percentage value.
+        value: Uint128,
```

Comment:
I think we can use [`Decimal`]([from_ratio](https://docs.rs/cosmwasm-std/0.9.2/cosmwasm_std/struct.Decimal.html#method.from_ratio)), instead.